src/main/kotlin/com/soywiz/korge/intellij/listeners/KorgeModuleRootListener.kt

```diff
@@ -0,0 +1,25 @@
+package com.soywiz.korge.intellij.listeners
+
+import com.intellij.openapi.roots.ModuleRootEvent
+import com.intellij.openapi.roots.ModuleRootListener
+import com.soywiz.korge.intellij.util.fixLibraries
+import java.util.concurrent.atomic.AtomicBoolean
+
+class KorgeModuleRootListener : ModuleRootListener {
+
+    private var updateInProgress = AtomicBoolean(false)
+
+    override fun beforeRootsChange(event: ModuleRootEvent) {
+    }
+
+    override fun rootsChanged(event: ModuleRootEvent) {
+        println("Running KorgeGradleSyncListener")
+        if (updateInProgress.get()) {
+            println("Update currently in progress... Quick returning.")
+            return
+        }
+        updateInProgress.set(true)
+        fixLibraries(event.project)
+        updateInProgress.set(false)
```

Comment:
To avoid inconsistencies if for some reasons there is an error:

```suggestion
        updateInProgress.set(true)
        try {
          fixLibraries(event.project)
        } finally {
          updateInProgress.set(false)
        }
```